### ISSUE TITLE
Project update

### ISSUE THREAD

**Issue by theofidry** [paragonie/pharaoh#16]
Hi,

This is just a message to say as time went on, I needed a better integration of Pharaoh withing Box. It was a bit awkward to do as pharaoh needed to be updated separately and also offers a standalone API. Since I was not too sure where to go with it exactly and the project was not really maintained, I made the decision to copy Pharaoh within Box (I kept the license for the relevant files, even there are barely recognizable now).

This lead to a huge refactoring effort which ultimately lead to:

- `KevinGH\Box\Phar\PharMeta` which represents a partial information of the `Phar|PharData` "metadata" (as not `Phar::metadata` the overall archive metadata) which does not require any code to be loaded (hence no process pollution neither alias collision).
- `KevinGH\Box\Phar\PharFactory` which is about creating a `Phar|PharData` instance of an _existing_ file (i.e. not accidentally create one) with more human-readable errors upon failure.
- `KevinGH\Box\Phar\PharInfo` which generates a `PharMeta` instance but leveraging Box `Extract` command to get all the data from the `Phar|PharData` instance _in a separate process_. This allows to circumvent some issues such as not having a `phar.readonly=1` issue, having to change the alias or other (hence the exact signature of the original PHAR can be retrieved).
- Ultimately some of the API could be extracted into a separate library if there is an interest in it. If one wanted to bring back Pharaoh to life I think this would make sense, although I would struggle to see why not using Box instead. IMO if there is more work to do it is within the PHP source itself.

**TL:DR;** I believe in the current state `box diff` is a better alternative to Pharaoh.

So AFAIC, this is where Box parts with Pharaoh as it is. It helped tremendously for years and this year work would definitely not have been possible without this project. For this, a big thanks, I learned a lot from it. 🙇 👏

**Comment by paragonie-security** [comment]
Your decision is 100% understandable and we look forward to how Box continues to develop in the future.

> TL:DR; I believe in the current state `box diff` is a better alternative to Pharaoh.

We're inclined to agree, and happy to deprecate this project and point folks to Box instead.
